internal/function/uast.go

```diff
@@ -1,9 +1,13 @@
 package function
 
 import (
+	"crypto/sha1"
 	"fmt"
+	"hash"
 	"strings"
 
+	lru "github.com/hashicorp/golang-lru"
```

Comment:
I thought about: `github.com/golang/groupcache/lru`, but if hashicorp works for you that's ok.

Reply:
It seems that groupcache/lru is not thread safe.

> Cache is an LRU cache. It is not safe for concurrent access.

[groupcache/lru godoc](https://godoc.org/github.com/golang/groupcache/lru)